chain/network/src/config.rs

```diff
@@ -446,7 +447,24 @@ impl NetworkConfig {
         self.routing_table_update_rate_limit
             .validate()
             .context("routing_table_update_rate_limit")?;
-        Ok(VerifiedConfig { node_id: self.node_id(), inner: self })
+
+        // Verify ip address can be signed and verified
+        if !(self.node_addr.is_some()) {
+            anyhow::bail!("no node address given that is necessary for near network handshake");
+        }
+        let owned_ip_address = OwnedIpAddress {
+            ip_address: self.node_addr.unwrap().ip(),
+        };
+        let my_node_id = self.node_id();
+        let my_signed_owned_ip_address: SignedOwnedIpAddress = owned_ip_address.sign(&self.node_key);
+        if !(my_signed_owned_ip_address.verify(my_node_id.public_key())) {
+            anyhow::bail!("Unable to sign and verify ip address with given node_key");
```

Comment:
Why would this ever happen? I don't think it makes sense to test this here.

Reply:
Do you have an example of a value which would trigger this check? My understanding is that we can sign an arbitrary sequence of bytes, so I don't understand the purpose of this section of code.

Additionally, take a look at the other checks performed here. They are examining simple boundary conditions on the config (for example, `ideal_connections_lo` should be <= `ideal_connections_hi`). They don't test implementation-specific details but rather basic properties of the config.

Reply:
This discussion is redundant as it's removed from latest code since we'll now use the tcp connection's address instead of configured listening address as mentioned in another comment.